pkg/util/cmd/split_args.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"strings"
+	"unicode"
+)
+
+// This is a '\'
+const escapeRune = rune(92)
+
+// SplitCmdString takes a single line command string and splits it into individual arguments to be used with
+// exec.Command. It does so by honouring quotes within the original command
+func SplitCmdString(cmd string) []string {
+	lastQuote := rune(0)
+	consecutiveEscapeCount := 0
+	splitWithQuotes := func(c rune) bool {
+		switch {
+		// don't split if the character matches the "last matched quote"
+		case c == lastQuote:
+			if consecutiveEscapeCount%2 == 0 {
+				lastQuote = rune(0)
+			}
+			consecutiveEscapeCount = 0
+			return false
+		// found an escapce character, don't split
+		case c == escapeRune:
+			consecutiveEscapeCount += 1
+			return false
+		// don't split if we are still in a quoted string
+		case lastQuote != rune(0):
+			consecutiveEscapeCount = 0
+			return false
+		// don't split if the character is a quote type rune, and set the lastQuote value to that
+		case unicode.In(c, unicode.Quotation_Mark):
+			lastQuote = c
+			consecutiveEscapeCount = 0
+			return false
+		// split the string if it is a space
```

Comment:
Not sure if it is worth noting in this comment, but maybe include something along the lines of "... and none of the other conditionals are satisfied".   That said, these are all very concise comments, so maybe leave it as is if that was the goal (I myself tend to get a little lengthy... character flaw...).